iothub/device/src/InternalClient.cs

```diff
@@ -621,7 +697,14 @@ public Task SendEventBatchAsync(IEnumerable<Message> messages, CancellationToken
             }
 
             // Codes_SRS_DEVICECLIENT_28_019: [The async operation shall retry until time specified in OperationTimeoutInMilliseconds property expire or unrecoverable error(authentication or quota exceed) occurs.]
-            return InnerHandler.SendEventAsync(messages, cancellationToken);
+            try
+            {
```

Comment:
Can you not do this before calling sendeventasync?

cancellationToken.ThrowIfCancellationRequested();

Reply:
No you can't. But it's a good idea add  `cancellationToken.ThrowIfCancellationRequested()` once cause is operation cancelled otherwise will just throw exception as it is.


`